### PR TITLE
a couple sort junits

### PR DESCRIPTION
**Pull request by eeckstrand**
any update on these issues?

-double sorting
-can't sort a frame if it has a string column, even if the string columns isn't a sort-by column

Thanks!
Eric

---

This change is [Reviewable](https://reviewable.io/reviews/h2oai/h2o-3/251)

### CODE REVIEW

File: h2o-core/src/test/java/water/rapids/SortTest.java [hunk 1]
```diff
@@ -67,6 +69,40 @@
...
+  }
+
+  @Test public void testBasicSortStringCol() {
+    Frame fr = null, sortedFr = null;
+    try {
+      Key raw = Key.make("sort_test_data_raw");
+      Key parsed = Key.make("sort_test_data_parsed");
+      FVecTest.makeByteVec(raw, "x,y\n0,a\n1,b\n2,c\n3,d\n4,e\n5,f\n6,g\n7,h");
+      fr = ParseDataset.parse(parsed, raw);
+      sortedFr = fr.sort(new int[]{0});
```
Comment (vpatryshev) [inline h2o-core/src/test/java/water/rapids/SortTest.java]: I wonder how come there are no assertions here.

**Comment by cliffclick** [comment]
Any progress on this?  Basically, these JUnits fail because the underlying sort either
- fails to sort doubles correctly, or
- crashes on strings

And now I've got an example which triggers an internal sort assertion.  Working on peeling out the dataset still, but basically I'm asking: what's the status of H2O's big sort?
Thanks
Cliff

**Comment by mmalohlava** [comment]
@h2o-ops please test

**Comment by mmalohlava** [comment]
@h2o-ops please re-build

**Comment by mmalohlava** [comment]
@h2o-ops please test

**Comment by mmalohlava** [comment]
@h2o-ops please test

**Comment by mmalohlava** [comment]
@h2o-ops please test

**Comment by mmalohlava** [comment]
@h2o-ops please test

**Comment by h2o-ops** [comment]
Tests passed!

**Comment by ledell** [comment]
@vpatryshev @mattdowle @cliffclick  What's the status on this?  It would be great to add support for sorting on floats.

**Comment by ledell** [comment]
@mattdowle There is a ticket for "does not sort on floats" here: https://0xdata.atlassian.net/browse/PUBDEV-3525

**Comment by abal5** [comment]
@h2o-ops please test!

**Comment by vpatryshev** [comment]
The code hardly tests anything. It's just called 'test', but there is no validation of the results of sorting; why?

**Comment by wendycwong** [comment]
@h2o-ops please test!

**Comment by wendycwong** [comment]
I have since fixed both issues:

a. sorting float
b. sorting frames with string columns.  

I am going to just close this PR now.

**Comment by wendycwong** [comment]
I have since fixed both issues:

a. sorting float
b. sorting frames with string columns.

I am going to just close this PR now.

**Pull request closed without merging**